lib/improver/utilities/cube_manipulation.py

```diff
@@ -464,33 +476,63 @@ def _equalise_coord_bounds(cubes):
 
     Raises:
         ValueError:
-            If two cubes with different valid bounds are found.
+            If some but not all cubes have bounds on a shared dimension
+            coordinate.
+        ValueError:
+            If existing bounds values on shared dimension coordinates do not
+            match.
     """
     # Check each cube against all remaining cubes
+    msg = 'Cubes with mismatching {} bounds are not compatible'
     for i, this_cube in enumerate(cubes):
         for later_cube in cubes[i+1:]:
-            for coord in this_cube.coords():
+            for coord in this_cube.coords(dim_coords=True):
                 try:
                     match_coord = later_cube.coord(coord)
                 except CoordinateNotFoundError:
                     continue
                 if coord.bounds is None and match_coord.bounds is None:
                     continue
                 elif coord.bounds is None and match_coord.bounds is not None:
-                    msg = ('Cubes with mismatching bounds are not '
-                           'compatible')
-                    raise ValueError(msg)
+                    raise ValueError(msg.format(coord.name()))
                 elif coord.bounds is not None and match_coord.bounds is None:
-                    msg = ('Cubes with mismatching bounds are not '
-                           'compatible')
-                    raise ValueError(msg)
-                elif np.allclose(np.array(coord.bounds),
-                                 np.array(match_coord.bounds)):
-                    continue
+                    raise ValueError(msg.format(coord.name()))
                 else:
-                    msg = ('Cubes with mismatching bounds are not '
-                           'compatible')
-                    raise ValueError(msg)
+                    if np.allclose(np.array(coord.bounds),
+                                   np.array(match_coord.bounds)):
+                        continue
+                    else:
+                        raise ValueError(msg.format(coord.name()))
+
+
+def _equalise_bounds_ranges(cube, coord_list):
+    """
+    Check the bounds ranges on a given list of coordinates match at each
+    point along that dimension.  For example: to check time and forecast period
+    ranges for accumulations to avoid blending 1 hr with 3 hr accumulations.
+
+    Args:
+        cube (iris.cube.Cube):
+            Input cube with dimensional coordinates in coord_list
```

Comment:
Should we add that the function assumes that the units on the coords are identical?

Reply:
I don't think it does.  For each coordinate, it loops over the points along that coordinate and checks all the bounds ranges are the same.  So it assumes every point on that coordinate has the same unit, but not that different coordinates have the same units, as it never compares points on different coordinate axes.

Reply:
Oh, yes. So it does. I had thought it was going through a list of cubes rather than a list of coords.